test/end-to-end/getRedirectionURL.test.js

```diff
@@ -377,5 +375,152 @@ describe("getRedirectionURL Tests", function () {
+                    await page.evaluateOnNewDocument(() => {
+                        localStorage.setItem("disableRedirectionAfterSuccessfulSignInUp", "true");
+                        localStorage.removeItem("isNewUserCheck");
+                    });
+                    await clearBrowserCookiesWithoutAffectingConsole(page, []);
+                    await Promise.all([
+                        page.goto(
+                            `${TEST_CLIENT_BASE_URL}/auth?authRecipe=passwordless&passwordlessContactMethodType=EMAIL`
+                        ),
+                        page.waitForNavigation({ waitUntil: "networkidle0" }),
+                    ]);
+                    await setPasswordlessFlowType("EMAIL", "USER_INPUT_CODE");
+                });
+
+                after(async function () {
+                    // Dont cleanup if tests were skipped
+                    if (didSkip) {
+                        return;
+                    }
+                    await browser.close();
+                    await fetch(`${TEST_SERVER_BASE_URL}/after`, {
+                        method: "POST",
+                    }).catch(console.error);
+                    await fetch(`${TEST_SERVER_BASE_URL}/stopst`, {
+                        method: "POST",
+                    }).catch(console.error);
+
+                    return screenshotOnFailure(this, browser);
+                });
+
+                it("should not do any redirection after successful sign up", async function () {
+                    await Promise.all([
+                        page.goto(`${TEST_CLIENT_BASE_URL}/auth`),
+                        page.waitForNavigation({ waitUntil: "networkidle0" }),
+                    ]);
+                    await setInputValues(page, [{ name: "email", value: exampleEmail }]);
+                    await submitForm(page);
+                    await waitForSTElement(page, "[data-supertokens~=input][name=userInputCode]");
+
+                    const urlBeforeSignUp = await page.url();
+
+                    const loginAttemptInfo = JSON.parse(
+                        await page.evaluate(() => localStorage.getItem("supertokens-passwordless-loginAttemptInfo"))
+                    );
+                    const device = await getPasswordlessDevice(loginAttemptInfo);
+                    await setInputValues(page, [{ name: "userInputCode", value: device.codes[0].userInputCode }]);
+                    await submitForm(page);
+                    // wait until network idle to ensure that the page has not been redirected
+                    await page.waitForNetworkIdle();
+
+                    const urlAfterSignUp = await page.url();
+                    const newUserCheck = await page.evaluate(() => localStorage.getItem("isNewUserCheck"));
+                    assert.equal(newUserCheck, "passwordless-true");
+                    assert.equal(urlBeforeSignUp, urlAfterSignUp);
```

Comment:
at this point, what does the UI on the screen look like - is it just the enter OTP screen UI only?

I ask cause we have [another UI state](https://supertokens.com/docs/passwordless/pre-built-ui/further-reading/passwordless-login#logging-in-via-otp-and-magic-link-simultaneously), which we should not be seeing in this case.